src/bokeh/models/sources.py

```
@@ -252,6 +252,13 @@ def column_names(self) -> list[str]:
         '''
         return list(self.data)
 
+    @property
+    def length(self) -> int:
+        ''' Number of row entries in the data. Note: All columns have the same number of row entries.
+
+        '''
+        return len(self.data[next(iter(self.data))]) if self.data else 0
```

So I would suggest doing this differently, by constructing a set with all the column lengths added. Then erroring if the set size is not one. Otherwise return the one length. Equal column lengths is a very strong assumption and should not be compromised. But in case it happens to be, I would rather raise a loud error than silently return a mis-matched length.

---

@bryevdv Gotcha, made some adjustments to account for that.